src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/trait/BitbucketLegacyTraitConverter.java

```diff
@@ -28,6 +29,10 @@ public static SCMSourceTrait maybeConvert(SCMSourceTrait trait) {
             return null;
```

Comment:
I can't comment outside of the diff, but the whole "throw exception in a private constructor" is not a pattern I would want us to have in this codebase. Just have a comment saying `//This is a utility class and should not be instantiated` and leave it at that. No need to change in this PR but before we merge to master it would be good to eradicate all such exceptions.

Reply:
Sounds good. I can update it as part of this PR.